cvmfs/util/algorithm.cc

```diff
@@ -105,6 +105,33 @@ std::vector<atomic_int32> UTLog2Histogram::GetBins(const Log2Histogram &h) {
   return h.bins_;
 }
 
+unsigned int Log2Histogram::q(float n) {
+  // we start by counting all the elements of the array, we include the overflow
+  uint64_t total = 0;
+  unsigned int i = 0;
+  for (i = 0; i <= this->bins_.size() - 1; i++) {
+    total += (unsigned int)atomic_read32(&(this->bins_[i]));
+  }
+  // now we found the index of the element we want
```

Comment:
Suggestion: `Find the index of the element corresponding to the requested quantile`